src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -844,17 +891,23 @@ private EventPipeProvider[] GetEventPipeProviders()
                 new("System.Diagnostics.Metrics", EventLevel.Informational, TimeSeriesValues,
                     new Dictionary<string, string>()
                     {
-                        { "SessionId", _metricsEventSourceSessionId },
+                        { "SessionId", SharedSessionId },
                         { "Metrics", metrics.ToString() },
                         { "RefreshInterval", _interval.ToString() },
                         { "MaxTimeSeries", _maxTimeSeries.ToString() },
-                        { "MaxHistograms", _maxHistograms.ToString() }
+                        { "MaxHistograms", _maxHistograms.ToString() },
+                        { "UniqueIdentifier", _uniqueIdentifier  }
                     }
                 );
 
             return eventCounterProviders.Append(metricsEventSourceProvider).ToArray();
         }
 
+        private bool Filter(string meterName, string instrumentName)
```

Comment:
Added a filter on the `dotnet counters` side similar to the one used by `dotnet monitor`; this weeds out things that this particular session might not care about that other SHARED sessions have requested.